cluster_utils/job.py

```diff
@@ -103,15 +102,19 @@ def generate_execution_cmd(self, paths, cmd_prefix: Optional[str] = None):
         set_cwd = "cd {}".format(paths["main_path"])
 
         if "variables" in paths:
-            if not isinstance(paths, dict):
+            if not isinstance(paths["variables"], dict):
                 raise ValueError(
                     'Expected type dict for "variables", but got type'
                     f' {type(paths["variables"])} instead'
                 )
+            env_variables = {
+                str(name): str(value) for name, value in paths["variables"].items()
+            }
             set_env_variables = "\n".join(
-                f"export {name}={value}" for name, value in paths["variables"].items()
+                f"export {name}={value}" for name, value in env_variables.items()
```

Comment:
When touching this anyway: I think it would be good to put quotes around `{value}`, otherwise values that contain spaces will be messed up.